### PR TITLE
simplify keycloak build script

### PR DESCRIPTION
**Pull request by npalaska**
Simplifies the Keycloak build script a bit more

- Remove the need for 2 redundant loops from the script
- Remove the need to check health uri when building OIDC keycloak server
- Quite the keycloak output down

### CODE REVIEW

**Review by dbutenhof — APPROVED** [review]
Still perhaps a bit wordy, (I suspect we don't really need all the horizontal lines), but much improved! I'll take it. 👍🏻 

```
Waiting for the Keycloak server

Keycloak connection successful on : http://localhost:8090/

Creating pbench realm
--------------
Creating pbench-server client
---------------
Getting client secret
---------------------
Creating an 'ADMIN' role under pbench-server client of the pbench realm
--------------------
Creating an 'admin' user inside pbench realm
-------------
Assigning 'ADMIN' client role to the user 'admin' created above
---------------------------
The Keycloak configuration is complete.        
```

File: server/pbenchinacan/load_keycloak.sh [hunk 1]
```diff
@@ -24,132 +24,112 @@ ADMIN_PASSWORD=${ADMIN_PASSWORD:-"admin"}
...
+    -d 'grant_type=password' \
+    -d 'client_id=admin-cli' | jq -r '.access_token')
+  if [[ ! -z "${ADMIN_TOKEN}" ]]; then
+    break
   else
-    echo "Keycloak health status code ${status_code}" >&2
+    echo "Waiting for the Keycloak server" >&2
   fi
   sleep 2
 done
```
Comment (webbnh) [inline server/pbenchinacan/load_keycloak.sh:49]: Having the time-out check (lines 33-36) at the top of the loop is...undesirable.  How about something more like this:
```
while true; do
  ADMIN_TOKEN=$(curl -s -f -X POST "${KEYCLOAK_HOST_PORT}/realms/master/protocol/openid-connect/token" \
    -H "Content-Type: application/x-www-form-urlencoded" \
    -d "username=${ADMIN_USERNAME}" \
    -d "password=${ADMIN_PASSWORD}" \
    -d 'grant_type=password' \
    -d 'client_id=admin-cli' | jq -r '.access_token')
  if [[ -n "${ADMIN_TOKEN}" ]]; then
    break
  elif [[ $(date +%s) -ge ${end_in_epoch_secs} ]]; then
    echo "Timed out connecting to Keycloak" >&2
    exit 1
  else
    echo "Waiting for the Keycloak server" >&2
  fi
  sleep 2
done
```

**Review by dbutenhof — APPROVED** [review]

File: server/pbenchinacan/load_keycloak.sh [hunk 2]
```diff
@@ -24,132 +24,112 @@ ADMIN_PASSWORD=${ADMIN_PASSWORD:-"admin"}
...
-
-echo "ADMIN_TOKEN=${ADMIN_TOKEN}"
+echo "Keycloak connection successful on : ${KEYCLOAK_HOST_PORT}"
 echo
 
-echo "Creating ${REALM} realm"
-echo "--------------"
-
-status_code=$(curl -f -i -o /dev/null -w "%{http_code}" -X POST "${KEYCLOAK_HOST_PORT}/admin/realms" \
+status_code=$(curl -f -si -o /dev/null -w "%{http_code}" -X POST "${KEYCLOAK_HOST_PORT}/admin/realms" \
```
Comment (webbnh) [inline server/pbenchinacan/load_keycloak.sh:55]: Do you really want to specify `-i` given that `-w`?

Ditto for the other places where you specify both.

**Review by webbnh — DISMISSED** [review]
Looks good; I just have a few suggestions.

File: server/pbenchinacan/load_keycloak.sh (hunk 2 above)
Comment (npalaska) [reply server/pbenchinacan/load_keycloak.sh:55]: Okay so `w` will write out the HTTP status code from the HTTP transfer on stdout and `i` will include all the headers. You are right `i` doesn't buy us anything when we specify `-w "%{http_code}"`

File: server/pbenchinacan/load_keycloak.sh [hunk 3]
```diff
@@ -24,59 +24,38 @@ ADMIN_PASSWORD=${ADMIN_PASSWORD:-"admin"}
 REALM=${KEYCLOAK_REALM:-"pbench"}
 CLIENT=${KEYCLOAK_CLIENT:-"pbench-server"}
 
-keycloak_health_uri="${KEYCLOAK_HOST_PORT}/health"
 end_in_epoch_secs=$(( $(date +%s) + 120 ))
```
[the remark this answers is not in the document]
Comment (npalaska) [reply server/pbenchinacan/load_keycloak.sh]: This is great, easier to read for sure!

**Review by webbnh — APPROVED** [review]
👍

**Review by dbutenhof — APPROVED** [review]
My last act this week; but I like this.

```
Keycloak connection successful on : http://localhost:8090/

Created pbench realm
Created pbench-server client
Created an 'ADMIN' role under pbench-server client of the pbench realm
Created an 'admin' user inside pbench realm
Assigned an 'ADMIN' client role to the user 'admin' created above
The Keycloak configuration is complete.
```

**Comment by webbnh** [comment]
> My last act this week

"Last" is just a [judge-y word](https://github.com/distributed-system-analysis/pbench/pull/3229#pullrequestreview-1283920561).  🤣
